boltdesignsystem/bolt
# Add new tertiary variant to the Button component

## Jira
http://vjira2:8080/browse/BDS-1465?filter=-1

## Summary
Adds a new semi-transparent "tertiary" variant of the Button, including psuedo states for hover, active, focus, disabled, etc + full theming support.

![CleanShot 2020-03-02 at 08 17 15](https://user-images.githubusercontent.com/1617209/75679838-61759b00-5c5e-11ea-9e30-2cf2e0767f0c.png)

## Details
- Manually specs out the `tertiary` theming system values to reduce the amount of cruft getting auto-generated --> likely a similar direction we're going to want to be going with other UI work moving forward
  - Note: since I'm _not_ using the `bolt-theme` Sass function here, I'm manually including basic IE 11 theming fallbacks styles (but can be easily removed in the near future)
- As a first for Bolt, tertiary Buttons now include basic theming support when _only_ the Button component's CSS / JS is compiled / loaded (unlike other Themed components which are visually broken if the `@bolt/global` styles aren't loaded).

<img width="1680" alt="CleanShot 2020-02-28 at 09 37 55@2x" src="https://user-images.githubusercontent.com/1617209/75561415-fa18da80-5a14-11ea-83eb-45a81824d36c.png">

### Button-related Cleanup
- Nixes some unused border-color CSS styles in Button since
  1. these effectively aren't visible since border color values are basically set to the background color 100% of the time and
  2. Can't be transparent (as I found out the hard way with Tertiary button) since they would otherwise be visible against the semi-transparent background

- Cleans up Button styles handling disabled states to be less complex + not require even more hacks in order to get Tertiary buttons styled correctly

## How to test
- Review code changes
- Confirm Button demos are visually and functionality working as expected
- Check out the new Tertiary styles (hover, active, etc -- on both light and dark themes) since I had very limited design comps to go off of for these!
